### PR TITLE
Add post how to properly stop a batch job execution or step execution

### PR DESCRIPTION
**Pull request by chengfang**
This post describes ways to stop batch job execution and step executions, and their implications, based on recent community discussion. No new features are involved and not time sensitive.

### CODE REVIEW

File: _posts/2020-11-01-batch-processing-stop-job-step.adoc [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    @Override
+    public String process() throws Exception {
+        String exitStatus = "BATCHLET1_COMPLETED";
+        while (true) {
+            if (toStop.get()) {
+                exitStatus = "BATCHLET1_STOPPED";
+                break;
+            }
+            // perform batchlet task
+            // Thread.sleep(5000);
```
Comment (jamezp) [inline _posts/2020-11-01-batch-processing-stop-job-step.adoc:56]: Should this be commented out?

File: _posts/2020-11-01-batch-processing-stop-job-step.adoc [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+{
+    "outcome" => "success",
+    "result" => 4L
+}
+----
+
+You can also perform all the above operations in WildFly Management Console. For example, the following
+screenshot shows the UI to stop a job execution:
+
+image::jberet/jberet-stop-job.png[Stop Batch Job Execution]
```
Comment (jamezp) [inline _posts/2020-11-01-batch-processing-stop-job-step.adoc:122]: Something is wrong with image resolution in general. I'm going to have a look at this as other posts are also missing images.

Comment (jamezp) [reply _posts/2020-11-01-batch-processing-stop-job-step.adoc:122]: I found the issue with this and filed https://github.com/wildfly/wildfly.org/issues/209. Nothing to change here as I'll send up a global fix.

**Comment by bstansberry** [comment]
@chengfang @jamezp Will this be moving forward? It needs a rebase and will need a date change.

**Comment by jamezp** [comment]
It looks like there is a conflict now so it needs a rebase regardless. The comments I made I guess shouldn't really block anything. They were just suggestions really.

**Comment by chengfang** [comment]
I'll resolve the conflit shortly.

File: _posts/2020-11-01-batch-processing-stop-job-step.adoc [hunk 3]
```diff
@@ -0,0 +1,229 @@
+---
+layout: post
+title: How to properly stop batch processing job and step in WildFly
+date: 2020-11-01
```
Comment (bstansberry) [inline _posts/2020-11-01-batch-processing-stop-job-step.adoc:4]: @chengfang Please change the 'date' field here to the day you make this change or the day after, and please also change the name of the file to match. Once it's updated please ping me in zulip so I can get it merged.

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: _posts/2020-11-01-batch-processing-stop-job-step.adoc [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+tags: wildfly jberet batch
+synopsis: How to properly stop a running batch job with API and WildFly tools, and how to stop an individual step.
+author: chengfang
+---
+
+Batch jobs are long-running background processing tasks, and therefore it's
+common that user may need to pause or stop the execution.  WildFly, which
+implements batch processing through its jberet component based on JSR 352
+and https://github.com/eclipse-ee4j/batch-api[Jakarta Batch Specification],
```
[the remark this answers is not in the document]
Comment (chengfang) [reply _posts/2020-11-01-batch-processing-stop-job-step.adoc]: fixed to use the official web site of eclipse bath project.

File: _posts/2020-11-01-batch-processing-stop-job-step.adoc [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+offers a variety of ways to achieve that requirement.  This post will
+demonstrate how to stop a running job execution, or individual step execution,
+and some design and implementation consideration.
+
+## Properly stop a running job execution
+
+Batch spec defines a standard API for stopping a running job execution:
+https://github.com/eclipse-ee4j/batch-api/blob/master/api/src/main/java/jakarta/batch/operations/JobOperator.java#L198[javax.batch.operations.JobOperator#stop]
```
[the remark this answers is not in the document]
Comment (chengfang) [reply _posts/2020-11-01-batch-processing-stop-job-step.adoc]: changed the link to EE9 javadoc of `JobOperator` class.

File: _posts/2020-11-01-batch-processing-stop-job-step.adoc (hunk 1 above)
Comment (chengfang) [reply _posts/2020-11-01-batch-processing-stop-job-step.adoc]: looks like the sleep line can be confusing. So I removed this sleep line and expand the //perform batchlet task to include some examples.

File: _posts/2020-11-01-batch-processing-stop-job-step.adoc (hunk 2 above)
Comment (chengfang) [reply _posts/2020-11-01-batch-processing-stop-job-step.adoc:121]: #209 has been resolved, so the image should work now.

File: _posts/2020-11-01-batch-processing-stop-job-step.adoc (hunk 3 above)
Comment (chengfang) [reply _posts/2020-11-01-batch-processing-stop-job-step.adoc]: changed the date field and renamed the file.

**Comment by bstansberry** [comment]
Thanks, @chengfang !

**Pull request merged**